staging/src/k8s.io/apimachinery/pkg/util/rand/rand.go

```diff
@@ -83,3 +83,13 @@ func String(length int) string {
 	}
 	return string(b)
 }
+
+// SafeEncodeString encodes s using the same characters as rand.String. This reduces the chances of bad words and
+// ensures that strings generated from hash functions appear consistent throughout the API.
+func SafeEncodeString(s string) string {
```

Comment:
Suggest putting this function elsewhere since the returned value isn't random

Reply:
It depends on the slice used random names generated by `String() string` in the same package. Its use is also closely tied to this function. It may warrant refactoring this code, but I'd rather focus on ensuring that the Names contain no bad words in this PR and a refactor in a followup.

Reply:
ok!